lightning/src/ln/channelmanager.rs

```diff
@@ -71,6 +72,40 @@ use core::sync::atomic::{AtomicUsize, Ordering};
 use core::time::Duration;
 use core::ops::Deref;
 
+/// A map with liquidity value (in msat) keyed by a short channel id and the direction the HTLC
+/// is traveling in. The direction boolean is determined by checking if the HTLC source's public
+/// key is less than its destination. See [`InFlightHtlcs::used_liquidity_msat`] for more
+/// details.
+#[cfg(not(any(test, feature = "_test_utils")))]
+pub struct InFlightHtlcs(HashMap<(u64, bool), u64>);
+#[cfg(any(test, feature = "_test_utils"))]
+pub struct InFlightHtlcs(pub HashMap<(u64, bool), u64>);
+
+impl InFlightHtlcs {
+	/// Create a new `InFlightHtlcs` via a mapping from:
+	/// (short_channel_id, source_pubkey < target_pubkey) -> used_liquidity_msat
+	pub fn new(inflight_map: HashMap<(u64, bool), u64>) -> Self {
+		InFlightHtlcs(inflight_map)
+	}
```

Comment:
Maybe could be left to a later refactor (cc: @jurvis), but it seems odd we expose the implementation details about the `HashMap`. Perhaps we should have a method for adding paths.

Another thought -- when adding this to the `Payer` interface -- is to have an associated type with a trait bound on some new trait having a `used_liquidity_msat`. Though that would mean `Router` and `Payer` would need to agree on that associated type as it would be need in `Router`, too.

Reply:
It does seem like the best way here could depend on how #1157 shakes out. I agree it's weird but I'm not 100% sure the best approach that wouldn't churn. Open to @jurvis's and your thoughts though if there seems to be a clear way. 

> when adding this to the Payer interface

Not following, what is being added to `Payer`?

Reply:
Yeah, I wouldn't be opposed to an associated type. I'm reluctant to be totally blocked on that issue though. Do you think there's something that can be done to make progress in the meantime here?

Reply:
Was leaning towards the former approach. Anyhow, this is orthogonal to the PR. Figured @jurvis would tackle it when updating the `Payer` trait.